lib/posix_channel.ml

```diff
@@ -196,8 +196,14 @@ let receive protocols =
   | V4V_proxy(_, _) -> assert false (* weight is 0 above *)
   | TCP_proxy(ip, port) ->
     let s = Unix.socket Unix.PF_INET Unix.SOCK_STREAM 0 in
-    Unix.connect s (Unix.ADDR_INET(Unix.inet_addr_of_string ip, port));
-    s
+    begin
+      try
+        Unix.connect s (Unix.ADDR_INET(Unix.inet_addr_of_string ip, port));
+        s
+      with e ->
```

Comment:
This likely looses the backtrace here

Reply:
This is a straight backport by cherry-picking. Do we want to do more?

Reply:
Lets backport this as is to Havana, and improve it on master.